collection/collection_map.go

```diff
@@ -102,22 +131,23 @@ func (c *Map) keys() []string {
 	return keys
 }
 
-// AddCS a value to some key with case sensitive vKey
-func (c *Map) AddCS(key string, vKey string, vVal string) {
-	aVal := types.AnchoredVar{Name: vKey, Value: vVal}
-	c.data[key] = append(c.data[key], aVal)
-}
-
 // Add a value to some key
 func (c *Map) Add(key string, value string) {
-	c.AddCS(key, key, value)
+	aVal := anchoredVar{Name: key, Value: value}
+	if !c.isKeySensitiveKey {
+		key = strings.AsciiLower(key)
+	}
+	c.data[key] = append(c.data[key], aVal)
 }
 
-// AddUniqueCS will add a value to a key if it is not already there
-// with case sensitive vKey
-func (c *Map) AddUniqueCS(key string, vKey string, vVal string) {
-	if c.data[key] == nil {
-		c.AddCS(key, vKey, vVal)
+// AddUnique will add a value to a key if it is not already there
+func (c *Map) AddUnique(key string, vVal string) {
```

Comment:
I only found one usage of addunique in our codebase

https://github.com/corazawaf/coraza/blob/63a49c8b17230f81a1a812298cbc0c336e59b15f/internal/corazawaf/transaction.go#L340

It doesn't seem to actually be needed there.

Can we leave it out?